doc/ntp/ntp-design.md

```diff
@@ -0,0 +1,898 @@
+4. `<key_id>` - Authentication key ID: `integer: 1-65535`
+
+_SRV_OPTS_:
+1. `-t|--type` - NTP server association type: `enum: server, pool`
+2. `-k|--key` - NTP authentication key ID: `integer: 1-65535`
+3. `-v|--version` - NTP protocol version `integer: 3-4`
+4. `-a|--aggressive` - NTP enable aggresive polling
+
+_KEY_OPTS_:
+1. `-t|--type` - NTP key type: `enum: md5, sha1`
+3. `-s|--trusted` - NTP key trustiness: `enum: yes, no`
+
+<!-- omit in toc -->
+### 2.6.2 Usage examples
+
+#### 2.6.2.1 Config command group
+
+**The following command enabled/disables NTP feature:**
+```bash
+config ntp state enabled
+config ntp state disabled
+```
+
+**The following command enabled/disables NTP authentication:**
+```bash
+config ntp authentication enabled
+config ntp authentication disabled
+```
+
+**The following command enabled/disables NTP DHCP client:**
+```bash
+config ntp dhcp enabled
+config ntp dhcp disabled
+```
+
+**The following command eonfigures NTP VRF:**
+```bash
+config ntp vrf default
+config ntp vrf mgmt
+```
+
+**The following command adds/deletes NTP server:**
+```bash
+config ntp server add 0.ua.pool.ntp.org \
+  --type server \
+  --key 42 \
+  --version 4
+config ntp server del 0.ua.pool.ntp.org
+```
+
+**The following command enabled/disables specific NTP server:**
+```bash
+config ntp server 0.ua.pool.ntp.org state enabled
+config ntp server 0.ua.pool.ntp.org state disabled
+```
+
+**The following command adds/removes NTP authentication key:**
+```bash
+config ntp key add 42 theanswer \
+  --type sha1 \
+  --trusted yes
+config ntp key del 42
+```
+
+#### 2.6.2.2 Show command group
+
+**The following command shows NTP state:**
+```
+root@sonic:/home/admin# show ntp
+MGMT_VRF_CONFIG is not present.
+synchronised to NTP server (10.10.10.2) at stratum 5
+   time correct to within 287 ms
+   polling server every 64 s
+     remote           refid      st t when poll reach   delay   offset  jitter
+==============================================================================
++10.10.10.1     162.159.200.123  4 u   53   64  377    0.134  +240.45  10.868
+*10.10.10.2     162.159.200.123  4 u   43   64  377    0.110  +231.47   5.604
+```
+
+**The following command shows NTP global config:**
+```
+root@sonic:/home/admin# show ntp global
+State: enabled
+Authentication: enabled
+VRF: default
+```
+
+**The following command shows NTP servers:**
+```
+root@sonic:/home/admin# show ntp server
+SERVER               STATE     TYPE    KEY  VERSION
+-------------------  --------  ------  ---  -------
+time.cloudflare.com  enabled   server       4
+2001:4860:4806::     enabled   server       3
+216.239.35.4         disabled  server  42   4
+pool.ntp.org         disabled  pool         4
+```
+
+**The following command shows NTP keys:**
+```
+root@sonic:/home/admin# show ntp key
+KEY ID  TYPE  VALUE       TRUSTED
+------  ----  ----------  -------
+1       md5   mypass      yes
+2       sha1  thebestntp  no
+42      md5   theanswer   yes
```

Comment:
The key cant be in plaintext, please fix it.

Reply:
ok